File: src/services/trading_service.py
class TradingService:
    def __init__(self):
        # Initialize inventory with some fruits
        self.inventory = {
            'apple': {'price': 1.0, 'quantity': 100},
            'banana': {'price': 0.5, 'quantity': 150},
            'orange': {'price': 0.75, 'quantity': 120}
        }

    def buy_fruit(self, fruit_name: str, quantity: int) -> str:
        fruit = self.inventory.get(fruit_name)
        if not fruit:
            return f"Fruit {fruit_name} does not exist."
        if fruit['quantity'] < quantity:
            return f"Not enough {fruit_name} in stock. Available: {fruit['quantity']}"
        fruit['quantity'] -= quantity
        return f"Bought {quantity} {fruit_name}(s). Remaining stock: {fruit['quantity']}"

    def sell_fruit(self, fruit_name: str, quantity: int) -> str:
        fruit = self.inventory.get(fruit_name)
        if not fruit:
            return f"Fruit {fruit_name} does not exist."
        fruit['quantity'] += quantity
        return f"Sold {quantity} {fruit_name}(s). New stock: {fruit['quantity']}"

    def get_inventory(self) -> dict:
        return self.inventory

File: src/services/tradingService.ts
import { Fruit } from '../types';

export class TradingService {
    private inventory: Map<string, { price: number; quantity: number }> = new Map();

    constructor() {
        // Initialize inventory with some fruits
        this.initializeInventory();
    }

    private initializeInventory(): void {
        this.inventory.set('apple', { price: 1, quantity: 100 });
        this.inventory.set('banana', { price: 0.5, quantity: 150 });
        this.inventory.set('orange', { price: 0.75, quantity: 120 });
    }

    public buyFruit(fruitName: string, quantity: number): string {
        const fruit = this.inventory.get(fruitName);
        if (!fruit) {
            return `Fruit ${fruitName} does not exist.`;
        }
        if (fruit.quantity < quantity) {
            return `Not enough ${fruitName} in stock. Available: ${fruit.quantity}`;
        }
        fruit.quantity -= quantity;
        return `Bought ${quantity} ${fruitName}(s). Remaining stock: ${fruit.quantity}`;
    }

    public sellFruit(fruitName: string, quantity: number): string {
        const fruit = this.inventory.get(fruitName);
        if (!fruit) {
            return `Fruit ${fruitName} does not exist.`;
        }
        fruit.quantity += quantity;
        return `Sold ${quantity} ${fruitName}(s). New stock: ${fruit.quantity}`;
    }

    public getInventory(): Map<string, { price: number; quantity: number }> {
        return this.inventory;
    }
}